### PR TITLE
[Core][MPI] Clean up `GatherModelPartUtility`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Clean up `GatherModelPartUtility`

**🆕 Changelog**

- [Clean up `GatherModelPartUtility`](https://github.com/KratosMultiphysics/Kratos/commit/9aa71225de91549b046ea99a438cff1344ea4d42)

### CODE REVIEW

File: kratos/mpi/python/add_mpi_utilities_to_python.cpp [hunk 1]
```diff
@@ -76,6 +76,7 @@ void AddMPIUtilitiesToPython(pybind11::module& m)
     ;
 
     py::class_<GatherModelPartUtility>(m, "GatherModelPartUtility")
+        .def(py::init<int, ModelPart&, int, ModelPart&>())
```
Comment (roigcarlo) [inline kratos/mpi/python/add_mpi_utilities_to_python.cpp]: ?

Comment (loumalouomega) [reply kratos/mpi/python/add_mpi_utilities_to_python.cpp]: The same constructor, sorry, I will correct

**Review by roigcarlo — COMMENTED** [review]
Aside from the duplicated constructor :+1:

**Review by roigcarlo — DISMISSED** [review]

File: kratos/mpi/utilities/gather_modelpart_utility.h [hunk 2]
```diff
@@ -4,66 +4,208 @@
...
+     */
+    GatherModelPartUtility(
+        const int GatherRank,
+        ModelPart& rOriginModelPart,
+        const int MeshId,
+        ModelPart& rDestinationModelPart
+        );
+
+    /// Destructor.
+    virtual ~GatherModelPartUtility() = default;
```
Comment (philbucher) [inline kratos/mpi/utilities/gather_modelpart_utility.h]: please remove this and let the compiler do its job 
=> then you also need to remove `virtual` from the other functions (anyway unused)

Comment (loumalouomega) [reply kratos/mpi/utilities/gather_modelpart_utility.h]: I added because the compiler require it

Comment (loumalouomega) [reply kratos/mpi/utilities/gather_modelpart_utility.h]: https://github.com/KratosMultiphysics/Kratos/actions/runs/5413385693/jobs/9838855668

Comment (philbucher) [reply kratos/mpi/utilities/gather_modelpart_utility.h]: I know, but there are virtual functions that do not need to be virtual
=> by removing the `virtual` you can also remove the DTor

**Review by philbucher — APPROVED** [review]

File: kratos/mpi/utilities/gather_modelpart_utility.h [hunk 3]
```diff
@@ -104,9 +104,6 @@ class KRATOS_API(KRATOS_MPI_CORE) GatherModelPartUtility
         ModelPart& rDestinationModelPart
         );
 
-    /// Destructor.
```
Comment (loumalouomega) [inline kratos/mpi/utilities/gather_modelpart_utility.h]: :D

**Pull request merged**